pkg/runtime/wfengine/backend.go

```diff
@@ -139,17 +140,33 @@ func (be *actorBackend) CreateOrchestrationInstance(ctx context.Context, e *back
 		workflowInstanceID = oi.GetInstanceId()
 	}
 
+	policy := &api.OrchestrationIdReusePolicy{}
+
+	for _, opt := range opts {
+		opt(policy)
+	}
+
 	eventData, err := backend.MarshalHistoryEvent(e)
 	if err != nil {
 		return err
 	}
 
+	createWorkflowInstanceRequest := CreateWorkflowInstanceRequest{
+		Policy:          policy,
+		StartEventBytes: eventData,
+	}
+
+	requestBytes, err := json.Marshal(createWorkflowInstanceRequest)
+	if err != nil {
+		return fmt.Errorf("failed to marshal createWorkflowInstanceRequest: %w", err)
+	}
+
 	// Invoke the well-known workflow actor directly, which will be created by this invocation
 	// request. Note that this request goes directly to the actor runtime, bypassing the API layer.
 	req := invokev1.
 		NewInvokeMethodRequest(CreateWorkflowInstanceMethod).
 		WithActor(be.config.workflowActorType, workflowInstanceID).
-		WithRawDataBytes(eventData).
+		WithRawDataBytes(requestBytes).
 		WithContentType(invokev1.OctetStreamContentType)
```

Comment:
nit: we should change this to specify that the payload is JSON. I don't think it actually matters from a runtime perspective, but it will at least help mitigate possible confusion.